lib/src/source_visitor.dart

```diff
@@ -76,14 +74,42 @@ class SourceVisitor extends ThrowingAstVisitor {
     return target is SimpleIdentifier && _looksLikeClassName(target.name);
   }
 
+  /// Whether [name] appears to be a type name.
+  ///
+  /// Type names begin with a capital letter and contain at least one lowercase
+  /// letter (so that we can distinguish them from SCREAMING_CAPS constants).
   static bool _looksLikeClassName(String name) {
     // Handle the weird lowercase corelib names.
     if (name == "bool") return true;
     if (name == "double") return true;
     if (name == "int") return true;
     if (name == "num") return true;
 
-    return _capitalPattern.hasMatch(name);
+    // TODO(rnystrom): A simpler implementation is to test against the regex
+    // "_?[A-Z].*?[a-z]". However, that currently has much worse performance on
+    // AOT: https://github.com/dart-lang/sdk/issues/37785.
+    const underscore = 95;
```

Comment:
I imagine you know about this package, but [charcode](https://pub.dev/packages/charcode) makes code like this readable.

Reply:
Yeah, I didn't feel like it was worth adding a dependency for five integers. :)